### PR TITLE
Drop pods can no longer land on death tiles (water, lava...)

### PR DESCRIPTION
**Pull request by XSlayer300**
## Changelog
:cl:
qol: After several incidents where drop pods accidentally land to places unsafe (*cough cough* water on Prison *cough*), TGMC finally upgraded the pod's safeties to not drop in these locations, as well as some locations they're not suppose to drop.
/:cl:

### CODE REVIEW

File: code/game/area/prison.dm [hunk 1]
```diff
@@ -191,6 +191,7 @@
 
 /area/prison/maintenance/residential/ne
 	name = "Northeast Civilian Residences Maintenance"
+	droppodable = FALSE
```
Comment (Alterist) [inline code/game/area/prison.dm:194]: this isn't needed with my recent PR

Comment (XSlayer300) [reply code/game/area/prison.dm]: Done.

File: code/game/area/area.dm [hunk 2]
```diff
@@ -41,6 +41,8 @@
 	var/fake_zlevel
 	///Is this area considered inside or outside
 	var/outside = TRUE
+	///Can you use this area to droppod?
+	var/droppodable = TRUE
```
Comment (TiviPlus) [inline code/game/area/area.dm:45]: This should be a new flag on flags_area instead

File: code/game/objects/structures/droppod.dm [hunk 3]
```diff
@@ -112,19 +114,25 @@
 /obj/structure/droppod/proc/checklanding(mob/user)
 	var/turf/target = locate(target_x, target_y, 2)
 	if(target.density)
-		to_chat(user, "<span class='warning'>Dense Landing zone detected. Invalid area.</span>")
+		to_chat(user, "<span class='warning'>Dense landing zone detected. Invalid area.</span>")
 		return FALSE
-	if(isspaceturf(target))
+	if(isspaceturf(target)) // Just a failsafe.
 		to_chat(user, "<span class='warning'>Location outside mission parameters. Invalid area.</span>")
 		return FALSE
```
Comment (TiviPlus) [inline code/game/objects/structures/droppod.dm:121]: Is this still needed?

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/droppod.dm (hunk 3 above)
Comment (XSlayer300) [reply code/game/objects/structures/droppod.dm:121]: Again, it is needed as a failsafe if it somehow fails.

Comment (TiviPlus) [reply code/game/objects/structures/droppod.dm:121]: But it wont, it has the flag?

**Comment by DerrickChamberlain** [comment]
Now we can finally put the walls back on prison.

File: code/game/area/general.dm [hunk 4]
```diff
@@ -10,6 +10,12 @@
 	temperature = TCMB
 	pressure = 0
 
+/area/space/Initialize(mapload)
+	. = ..()
+	if(mapload)
+		var/area/ourarea = get_area(src)
+		ENABLE_BITFIELD(ourarea.flags_area, NO_DROPPOD)
```
Comment (BraveMole) [inline code/game/area/general.dm:17]: no need. 

you can do the same but better by changing the var on compilation

/area/space
	name = "Space"
	requires_power = 1
	always_unpowered = 1
	dynamic_lighting = DYNAMIC_LIGHTING_ENABLED
	power_light = 0
	power_equip = 0
	power_environ = 0
	ambience = list('sound/ambience/ambispace.ogg')
	temperature = TCMB
	pressure = 0
    flags_area = NO_DROPPOD <------

Comment (TiviPlus) [reply code/game/area/general.dm:17]: Use codeblocks, but yea basically this

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**